stdlib/collections/__init__.pyi

```diff
@@ -69,7 +71,7 @@ class UserList(MutableSequence[_T]):
     @overload
     def __getitem__(self, i: int) -> _T: ...
     @overload
-    def __getitem__(self, i: slice) -> MutableSequence[_T]: ...
+    def __getitem__(self: _UserListT, i: slice) -> _UserListT[_T]: ...
```

Comment:
We actually have a fairly new `Self` alias for that, which you can import from `_typeshed`:

```python
from _typeshed import Self

[...]

@overload
    def __getitem__(self: Self, i: slice) -> Self: ...
```

Reply:
Oh that would be a very comfortable way. But does that work? `Self` from `_typeshed` is defined as `Self = TypeVar("Self")` (implicitly `bound=object`) and not `TypeVar("_UserListT", bound=UserList)` and the `UserList.__getitem__` implementation is accessing the attribute `self.data`, and the upper bound of `self` namely `object` does not have this attribute.

Reply:
A typevar annotating `self` is implicitly bound to that class. I'm not sure this is documented anywhere, but this is how type checkers work.

Reply:
At least I never read that, and hence did not know about that, … but that behaviour makes life easier. Thanks for explanation! I changed the PR to `_typesheld.Self`.